### PR TITLE
Opsrc 588/saving wishlist

### PR DESCRIPTION
**Pull request by milwoz**

### CODE REVIEW

File: src/CommandHandler/Wishlist/AddWishlistToUserHandler.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        $wishlist = $addWishlistsToUser->getWishlist();
+        $user = $addWishlistsToUser->getShopUser();
+        $wishlistCookieToken = $this->wishlistCookieTokenResolver->resolve();
+
+        if ($wishlistCookieToken !== $wishlist->getToken()){
+            throw new WishlistHasAnotherShopUserException();
+        }
+
+        if ($this->wishlistRepository->findOneByShopUserAndName($user, $wishlist->getName())) {
```
Comment (christopherhero) [inline src/CommandHandler/Wishlist/AddWishlistToUserHandler.php:43]: suggestion: can you check with returned type ? 
(i mean check `infance of`)

File: src/Controller/Action/AddWishlistToUserAction.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $wishlistId = $request->attributes->getInt('id');
+        $wishlist = $this->wishlistRepository->find($wishlistId);
+
+        /** @var ShopUserInterface $shopUser */
+        $shopUser = $this->tokenStorage->getToken()->getUser();
```
Comment (christopherhero) [inline src/Controller/Action/AddWishlistToUserAction.php:63]: are you sure the token always is not null ?

Comment (christopherhero) [reply src/Controller/Action/AddWishlistToUserAction.php:63]: can be moved to definition block

```
        /** @var ShopUserInterface $shopUser */
        $shopUser = $this->tokenStorage->getToken()->getUser();
        $wishlistId = $request->attributes->getInt('id');
        $wishlist = $this->wishlistRepository->find($wishlistId);

```

File: src/Controller/Action/ShowChosenWishlistAction.php [hunk 3]
```diff
@@ -56,15 +61,18 @@ public function __construct(
...
     public function __invoke(string $wishlistId, Request $request): Response
     {
         /** @var WishlistInterface $wishlist */
         $wishlist = $this->wishlistRepository->find((int)$wishlistId);
         $wishlistCookieToken = $this->wishlistCookieTokenResolver->resolve();
+        $user = $this->tokenStorage->getToken() ? $this->tokenStorage->getToken()->getUser() : null;
 
-        if ($wishlist instanceof WishlistInterface && $wishlist->getToken() === $wishlistCookieToken) {
+        if ($wishlist instanceof WishlistInterface && $user instanceof ShopUserInterface
+        or $wishlist instanceof WishlistInterface && $wishlist->getToken() === $wishlistCookieToken && $wishlist->getShopUser() === null) {
```
Comment (christopherhero) [inline src/Controller/Action/ShowChosenWishlistAction.php:75]: please change `or to ||`

Comment (christopherhero) [inline src/Controller/Action/ShowChosenWishlistAction.php:75]: line to long

**Review by christopherhero — CHANGES_REQUESTED** [review]

File: src/Controller/Action/ShowChosenWishlistAction.php (hunk 3 above)
Comment (milwoz) [reply src/Controller/Action/ShowChosenWishlistAction.php:75]: Done.

Comment (milwoz) [reply src/Controller/Action/ShowChosenWishlistAction.php:75]: Done.

File: src/CommandHandler/Wishlist/AddWishlistToUserHandler.php (hunk 1 above)
Comment (milwoz) [reply src/CommandHandler/Wishlist/AddWishlistToUserHandler.php:43]: Done.

File: src/Controller/Action/AddWishlistToUserAction.php (hunk 2 above)
Comment (milwoz) [reply src/Controller/Action/AddWishlistToUserAction.php:63]: Done.

**Review by christopherhero — APPROVED** [review]

**Pull request merged**